yt_dlp/extractor/mocha.py

```diff
@@ -0,0 +1,68 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, traverse_obj
+
+
+class MochaVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://video.mocha.com.vn/(?P<video_slug>[\w-]+)'
+    _TESTS = [{
+        'url': 'http://video.mocha.com.vn/chuyen-meo-gia-su-tu-thong-diep-cuoc-song-v18694039',
+        'info_dict': {
+            'id': '18694039',
+            'title': 'Chuyện mèo giả sư tử | Thông điệp cuộc sống',
+            'ext': 'mp4',
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'display_id': 'chuyen-meo-gia-su-tu-thong-diep-cuoc-song',
+            'thumbnail': 'http://mcvideomd1fr.keeng.net/playnow/images/20220505/ad0a055d-2f69-42ca-b888-4790041fe6bc_640x480.jpg',
+            'description': '',
+            'duration': 70,
+            'timestamp': 1652254203,
+            'upload_date': '20220511',
+            'comment_count': int,
+            'categories': ['Kids']
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_slug = self._match_valid_url(url).group('video_slug')
+        json_data = self._download_json(
+            'http://apivideo.mocha.com.vn:8081/onMediaBackendBiz/mochavideo/getVideoDetail',
+            video_slug, query={'url': f'{url}', 'token': ''})
+
+        video_url = (traverse_obj(json_data, ('data', 'videoDetail', 'list_resolution'))
+                     or traverse_obj(json_data, ('data', 'videoDetail', 'original_path')))
```

Comment:
```suggestion
        video_url = traverse_obj(json_data, (
            'data', 'videoDetail', ('list_resolution', 'original_path')), get_all=False)
```

Reply:
if we want to extract both, we should `get_all` right ?

Reply:
Yes, but that is default. So it'll be just:
```suggestion
        video_url = traverse_obj(json_data, (
            'data', 'videoDetail', ('list_resolution', 'original_path')))
```
or
```suggestion
        video_url = traverse_obj(json_data, (
            'data', 'videoDetail', (('list_resolution', ...), 'original_path')))
```
depending on how exactly to want to process it